src/main/java/edu/hm/hafner/coverage/parser/LcovParser.java

```diff
@@ -0,0 +1,194 @@
+package edu.hm.hafner.coverage.parser;
+
+import edu.hm.hafner.coverage.Coverage.CoverageBuilder;
+import edu.hm.hafner.coverage.CoverageParser;
+import edu.hm.hafner.coverage.FileNode;
+import edu.hm.hafner.coverage.Metric;
+import edu.hm.hafner.coverage.ModuleNode;
+import edu.hm.hafner.coverage.PackageNode;
+import edu.hm.hafner.coverage.Value;
+import edu.hm.hafner.util.FilteredLog;
+import edu.hm.hafner.util.LookaheadStream;
+import edu.hm.hafner.util.PathUtil;
+import org.apache.commons.lang3.tuple.MutablePair;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.Serial;
+import java.nio.file.Path;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * A parser for LCOV coverage reports.
+ */
+public class LcovParser extends CoverageParser {
+    @Serial
+    private static final long serialVersionUID = 1L;
+
+    private static final PathUtil PATH_UTIL = new PathUtil();
+
+    /**
+     * Creates a new instance of {@link LcovParser}.
+     */
+    public LcovParser() {
+        this(ProcessingMode.FAIL_FAST);
+    }
+
+    /**
+     * Creates a new instance of {@link LcovParser}.
+     *
+     * @param processingMode
+     *         determines whether to ignore errors
+     */
+    public LcovParser(final ProcessingMode processingMode) {
+        super(processingMode);
+    }
+
+    @Override
+    @SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.CognitiveComplexity"})
+    protected ModuleNode parseReport(final Reader reader, final String fileName, final FilteredLog log) {
+        final var root = new ModuleNode(EMPTY);
+        final Map<String, Map<Integer, MutablePair<Integer, Integer>>> files = new LinkedHashMap<>();
```

Comment:
```suggestion
       var root = new ModuleNode(EMPTY);
       Map<String, Map<Integer, MutablePair<Integer, Integer>>> files = new LinkedHashMap<>();
```

Reply:
If i'm not mistaken checkstyle was complaining about non-final vars.

But I will check again